docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
-If everything checks out, you're ready to get started on a new formula!
+If everything checks out, you're ready to get started on a new formula / cask!
+
+**Note:** For updating the version of an existing formula or cask, refer to the [Updating Software in Homebrew](Updating-Software-in-Homebrew.md) guidelines, as well as 'submitting a new version of a [formula](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-formula) / [cask](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-cask)' in 'How To Open A Homebrew Pull Request'.
 
 ## Formulae
 
+**Note:** Before taking the time to craft a new formula:
+
+* make sure it can be accepted by checking [Acceptable Formulae](Acceptable-Formulae.md).
+* check that the formula was not [already refused](https://github.com/Homebrew/homebrew-core/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing formula, see the streamlined method in ['submit a new version of an existing formula'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-formula).
+
 ### Writing the formula
 
-1. It's a good idea to find existing formulae in Homebrew that have similarities to the software you want to add. This will help you to understand how specific languages, build methods, etc. are typically handled. Start by tapping `homebrew/core`: first set `HOMEBREW_NO_INSTALL_FROM_API=1` in your shell environment, then run `brew tap homebrew/core` to clone the `homebrew/core` tap to the path returned by `brew --repository homebrew/core`.
+Making a new formula is easy, and the [Formula Cookbook](Formula-Cookbook.md) is an essential resource to guide you through the process. The Cookbook provides detailed instructions, best practices, and solutions to common problems you may encounter when creating a formula. Familiarizing yourself with its contents before you start is advisable.
 
-1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python`.
+1. Begin by researching existing formulae in Homebrew that are similar to the software you plan to add. This will help you grasp how specific languages, build methods, and other typical conventions are handled in formulae. Start by setting `HOMEBREW_NO_INSTALL_FROM_API=1` in your shell environment. Then, run `brew tap homebrew/core` to clone the `homebrew/core` tap to the directory indicated by `brew --repository homebrew/core`.
```

Comment:
Can you avoid rewording from e.g. "It's a good idea to" to "Begin by researching existing formulae"? It makes it much harder to distinguish semantic changes from wording preferences (that may or may not be better but are best addressed in a follow-up PR).

Reply:
@MikeMcQuaid Original wording restored in 038035cf664b2b1e3b7ddeb2faacc76c3540023b, cask wording changed to align with that in 4a5d9cf5a1271b0b9951293dff7a1ba1a8dfafe3

I'll leave this comment thread open for visibility for now; but once you check it and are happy, leave a comment and then feel free to mark it resolved.